### ISSUE TITLE
A way to diff that only recurses if outputs are different

### ISSUE THREAD

**Issue by ocharles** [Gabriel439/nix-diff#4]
I have two derivations that are different, `/nix/store/f6a70iijzdq7xzkyycqcggqdv2s77nla-HsOpenSSL-0.11.4.11.drv` and `/nix/store/7kmc4asp07ffp9ppihmvl0xjk6qcq2gq-HsOpenSSL-0.11.4.11.drv`. However, this doesn't actually matter, because the outputs of both of these derivations are the same thing. The differences in derivation are due to how tars are fetched (different mirrors), but the result is the same due to the fixed output.

Thus while these are different derivations, they don't affect substitution. It'd be nice if `nix-diff` didn't consider such derivations further (just check that the outputs are the same).

**Comment by Gabriel439** [comment]
Yeah, I just ran into this issue, too.  I agree that this should be fixed although I won't be able to get to it for a few weeks if that's alright

**Comment by expipiplus1** [comment]
Having this be an optional feature would be helpful. It could be useful to
be informed of differences even with fixed outputs.

**Comment by Gabriel439** [comment]
I have a quick and dirty branch up here:

https://github.com/Gabriel439/nix-diff/tree/gabriel/fixed_outputs

I'll test it more once I have another example of this at work to test drive this on
